docs/stargz-estargz.md

```diff
@@ -1,81 +1,223 @@
+- The end of the payload of each non-empty regular file entry
+- Arbitrary location within the payload of non-empty regular file entry
 
-eStargz also has content verification functionality based on OCI [_digests_](https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#digests).
-See the [verification documentation](/docs/verification.md) for more detail.
+The gzip header locations described in the second item MAY be used for chunking large regular files into several gzip members.
+Each chunked member is called *chunk* in this doc.
+An eStargz archive is the concatenation of these gzip members, which is a still valid gzip.
 
-### The structure of eStargz
+## TOC, TOCEntries and Footer
+
+### TOC and TOCEntries
+
+A regular file entry called *TOC* MUST be contained as the last tar entry in the archive.
+TOC MUST be a JSON file and MUST be named `stargz.index.json`.
+
+TOC records all file's metadata (e.g. name, file type, owners, offset etc) in the tar archive, except TOC itself.
+The TOC is defined as the following.
+
+- **`version`** *int*
+
+   This REQUIRED property contains the version of the TOC. This value MUST be `1`.
+
+- **`entries`** *array of objects*
+
+   Each item in the array MUST be a TOCEntry.
+   This property MUST contain TOCEntries that reflect all tar entries and chunks, except `stargz.index.json`.
+
+The TOCEntry is defined as the following.
+If the information written in TOCEntry differs from the corresponding tar entry, TOCEntry SHOULD be respected.
+TOCEntries fields other than `chunkDigest` are inherited from [stargz](https://github.com/google/crfs).
+
+- **`name`** *string*
+
+  This REQUIRED property contains the name of the tar entry.
+  This MUST be the complete path stored in the tar file.
+
+- **`type`** *string*
+
+  This REQUIRED property contains the type of the tar entry.
+  This MUST be either of the following.
+  - `dir`: directory
+  - `reg`: regular file
+  - `symlink`: symbolic link
+  - `hardlink`: hard link
+  - `char`: character device
+  - `block`: block device
+  - `fifo`: fifo
+  - `chunk`: a chunk of regular file data
+  As described in the above section, a regular file can be divided into several chunks.
+  Corresponding to the first chunk of that file, TOCEntry typed `reg` MUST be contained.
+  Corresponding to the chunks after 2nd, TOCEntries typed `chunk` MUST be contained.
+  `chunk`-typed TOCEntry must set offset, chunkOffset and chunkSize properties.
+
+- **`size`** *int64*
+
+  This OPTIONAL property contains the uncompressed size of the regular file tar entry.
+
+- **`modtime`** *string*
+
+  This OPTIONAL property contains the modification time of the tar entry.
+  Empty means zero or unknown.
+  Otherwize, the value is in UTC RFC3339 format.
+
+- **`linkName`** *string*
+
+  This OPTIONAL property contains the link target of `symlink` and `hardlink`.
+
+- **`mode`** *int64*
+
+  This OPTIONAL property contains the permission and mode bits.
+
+- **`uid`** *int*
```

Comment:
uint

Reply:
Fixed this. The implementation will be following-up PRs.